projects/plugins/jetpack/_inc/client/lib/jp-redirect/index.js

```diff
@@ -36,7 +41,7 @@ export default function getRedirectUrl( source, args = {} ) {
 		queryVars.source = encodeURIComponent( source );
 	}
 
-	const acceptedArgs = [ 'site', 'path', 'query', 'anchor' ];
+	const acceptedArgs = [ 'site', 'path', 'query', 'anchor', 'purchasetoken' ];
```

Comment:
Hi @elliottprogrammer ,

I'm looking only at how you are adding a new parameter to the Redirect URL.

We don't need to add a new parameter in the accepted Args there. The `query` arg is a generic parameter where you can add everything you need.

You have to pass it is as a string. For example: `key1=value1&key2=value2` and this will be added to the final URL after the user reaches the redirect server.

Reply:
Hi @leogermani,

You're right. I replaced the `purchasetoken` query parameter with `query` as you told us and it worked fine.

I pushed a new commit that adds this change and updated the branch against master.

Reply:
thanks!